libfreerdp/core/childsession.c

```diff
@@ -151,7 +151,7 @@ static int transport_bio_named_read(BIO* bio, char* buf, int size)
 
 			if (ptr->opInProgress)
 			{
-				DWORD status = WaitForSingleObjectEx(ptr->readEvent, 500, TRUE);
+				DWORD status = WaitForSingleObjectEx(ptr->readEvent, INFINITE, TRUE);
 				switch (status)
```

Comment:
I feel like we could get stuck forever when it closes. We should probably do something in `doReadOp` to set `readEvent`